packages/realm_dart/lib/src/cli/atlas_apps/baas_client.dart

```diff
@@ -243,25 +306,26 @@ class BaasClient {
   /// @nodoc
   Future<List<BaasApp>> getOrCreateApps() async {
     var apps = await _getApps();
-    await _createAppIfNotExists(apps, defaultAppName, _appSuffix);
-    await _createAppIfNotExists(apps, "autoConfirm", _appSuffix, confirmationType: "auto");
-    await _createAppIfNotExists(apps, "emailConfirm", _appSuffix, confirmationType: "email");
+    await _createAppIfNotExists(apps, AppNames.flexible);
+    await _createAppIfNotExists(apps, AppNames.autoConfirm);
+    await _createAppIfNotExists(apps, AppNames.emailConfirm);
+    await _createAppIfNotExists(apps, AppNames.staticSchema);
     return apps;
   }
 
   Future<void> waitForInitialSync(BaasApp app) async {
     while (!await _isSyncComplete(app.appId)) {
       print('Initial sync for ${app.name} is incomplete. Waiting 5 seconds.');
-      await Future.delayed(Duration(seconds: 5));
+      await Future<Object?>.delayed(Duration(seconds: 5));
     }
 
     print('Initial sync for ${app.name} is complete.');
   }
 
-  Future<void> _createAppIfNotExists(List<BaasApp> existingApps, String appName, String appSuffix, {String? confirmationType}) async {
-    final existingApp = existingApps.firstWhereOrNull((a) => a.name == appName);
+  Future<void> _createAppIfNotExists(List<BaasApp> existingApps, AppNames appName) async {
```

Comment:
```suggestion
  Future<void> _createAppIfNotExists(List<BaasApp> existingApps, AppName appName) async {
```
Looks odd here due to plural